city.lua

```diff
@@ -1092,6 +1074,25 @@ local function growCitizenCount(city, count, tier)
     Consumption.updateNeeds(city)
 end
 
+--- @param buildingTypes BuildingType[] | nil
+--- @return string tileType
+local function get_ground_tile_type(buildingType) 
+    -- "simple"
+    -- "residential"
+    -- "highrise"
+    -- "tycoon-treasury"
+    -- "garden"
+    if buildingType == "simple" then
+        return Constants.GROUND_TILE_TYPES.simple
+    elseif buildingType == "residential" then
+        return Constants.GROUND_TILE_TYPES.residential
+    elseif buildingType == "highrise" then
+        return Constants.GROUND_TILE_TYPES.highrise
+    else
+        return Constants.GROUND_TILE_TYPES.simple
+    end
```

Comment:
i don't get why you use these ugly string comparisons here again after you've added neat `GROUND_TILE_TYPES` table in `constants.lua` (below)

probably the only good thing about Lua is that it allows named field access both ways like this:
```
-- these are identical:
t = { a = 1, b = 2 }
t = { ["a"] = 1, ["b"] = 2 }
-- access/assignment is identical, too:
t.a, t.b
t["a"], t["b"]
```
don't remember where i've read that, though. that's language syntax, probably

so, this function actually becomes a one-liner:
```
return Constants.GROUND_TILE_TYPES[buildingType] or Constants.GROUND_TILE_TYPES.simple
```

Reply:
Yeah, that looks better! I'll change it.